pkg/kgo/client.go

```diff
@@ -483,7 +492,7 @@ func NewClient(opts ...Opt) (*Client, error) {
 		prsPool: newPrsPool(),
 
 		compressor:   compressor,
-		decompressor: newDecompressor(),
+		decompressor: newDecompressor(decompressorPool),
```

Comment:
#4 removes the decompressor from these fields and uses a global instance instead. I'm trying to think about how we can make these two changes work together. 

Can we change the decompressor to accept a `*pool.BucketedPool[byte]` on each call to `decompress` (or have another method: `decompressWithPool`)? Then we can also include a `DecompressedBytes *pool.BucketedPool[byte]` to `ProcessFetchPartitionOptions`?

Reply:
_addressed @ 8d36414e993ed377ebd0b897f7d441d962df5a85_